### DISCUSSION TITLE
How do I get the Bluetooth MAC Address of the device connected to my ESP32?

### DISCUSSION THREAD

**Discussion by ve2stn** [micropython/micropython#9648] [Hardware & Peripherals]
Hey guys I trying to understand the ESP-BT commands to figure out how I can pull the Bluetooth Address of the device connected via Bluetooth to my ESP32.

My goal is to save it so that next time I power up I will go searching for the device instead of having to pair it everytime...

Thanks Richard

**Comment by jimmo** [comment]
If you're a peripheral, then you'll receive the remote central's address in the connection event (`_IRQ_CENTRAL_CONNECT`).

But as a peripheral, you likely won't be able to reverse roles and connect to the other device -- most devices are typically one or the other. (unless you control both sides I guess?).

**Reply by ve2stn** [reply]
Thanks for the info, any suggestions how to go read that address???

commands related to that???

**Reply by jimmo** [reply]
I'm not sure what you mean sorry... It's passed to the event in the data tuple.

If you use `aioble` (https://github.com/micropython/micropython-lib/tree/master/micropython/bluetooth/aioble) then you can access `device.addr` or `device.addr_hex()`.
